src/browser-frontend/sagas/webcontents-action-handlers.js

```diff
@@ -94,8 +94,8 @@ function* onPageDidNavigateInternal() {
   // TODO
 }
 
-function* onPageDidNavigateToNewWindow() {
-  // TODO
+function* onPageDidNavigateToNewWindow({ payload: { pageId, url } }) {
+  yield put(PagesModelActions.addPage({ url, pageId }));
```

Comment:
Nit: it's useless to pass in the pageId here to the addPage action, which doesn't use it. That id is the id of the page where the navigation occured, and every single page/tab needs to have a unique id. The reducer handling this addPage action will create a new id for the newly added page/tab. Currently we're not tracking tab parents, but if we did remembering this old id here would be a good idea.